Test_Engine/Query/FullMessage.cs

```diff
@@ -0,0 +1,101 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Base;
+using BH.oM.Test;
+using BH.oM.Test.Results;
+
+namespace BH.Engine.Test
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Returns a full concatenated message for the test results and its inner result to the target max depth, only giving messages worse or equal to the provided severity level.")]
+        [Input("result", "The TestResult to give the full message for.")]
+        [Input("maxDepth", "Maximum level of recursiveness for inner TestInformation.")]
+        [Input("minSeverity", "The minimum level of severity of the information to report back.")]
+        [Output("message", "Full message for the TestResult")]
+        public static string FullMessage(this TestResult result, int maxDepth = 3, TestStatus minSeverity = TestStatus.Warning)
```

Comment:
Following our discussion, could we default to a `Pass` on the severity, so the default is to return everything? Or is there a reason for default to `Warning`? Can be happy to keep as is if there's a reason for it :smile:

Reply:
Happy to default to `Pass`. My reasoning for defaulting to warning was to default to only show anything problematic, but happy to change

Reply:
I feel like showing everything by default works well, with different applications choosing their level as appropriate?